### DISCUSSION TITLE
Request for adding an API to do retrieval from knowledge base

### DISCUSSION THREAD

**Discussion by sunxichen** [langgenius/dify#3286] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

Yes, some of my llm projects do RAG but without the last generation step (only retrieve some segments from knowledge base, and do some other processes). But, I want to use Dify to manage these knowledge bases. And there is no such APIs to directly achieve this goal.

### 2. Describe the feature you'd like to see

Add some APIs that can retrieve some segments from a knowledge base dataset using hybrid_search, full text search or semantic search method.

### 3. How will this feature improve your workflow or experience?

To achieve this goal, I have to use /console/api/datasets/{DATASET_ID}/hit-testing api for now. It is not direct method, and have to use user login token which is hard to manage.

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by chazzhou** [comment]
Second this suggestion. In my use case, my company has internal RAG solutions that can return text directly through restapi. It would be great if external contexts could be supported in an official dataprovider type way.
